flax/linen/recurrent.py

```diff
@@ -142,6 +142,146 @@ def initialize_carry(rng, batch_dims, size, init_fn=zeros):
     return init_fn(key1, mem_shape), init_fn(key2, mem_shape)
 
 
+class DummyDense(Module):
```

Comment:
I tried to make it private to `OptimizedLSTMCell` but Flax doesn't like that (error), so I'll keep this outside.

I renamed to `DenseParams` because that is what it is. Does that work for you? I still like `DummyDense` too though, since it follows the Dense API, just doesn't apply it.